src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -148,13 +150,15 @@ public OicSecurityRealm(String clientId, String clientSecret, String wellKnownOp
 
             this.authorizationServerUrl = config.getAuthorizationEndpoint();
             this.tokenServerUrl = config.getTokenEndpoint();
+            this.tokenAuthMethod = config.getPreferredTokenAuthMethod();
             this.userInfoServerUrl = config.getUserinfoEndpoint();
             this.scopes = config.getScopesSupported() != null && !config.getScopesSupported().isEmpty() ? StringUtils.join(config.getScopesSupported(), " ") : "openid email";
             this.logoutFromOpenidProvider = logoutFromOpenidProvider != null;
            	this.endSessionEndpoint = config.getEndSessionEndpoint();
         } else {
             this.authorizationServerUrl = authorizationServerUrl;
             this.tokenServerUrl = tokenServerUrl;
+            this.tokenAuthMethod = tokenAuthMethod;
```

Comment:
Wouldn't that break previous configuration ?
Default to "client_secret_post" if empty or null.